src/routes/api/client-key/client-key-regex.ts

```diff
@@ -1,4 +1,4 @@
 export const extractClientKey = (text: string): string | undefined  => {
-	const [, plainClientKey] = /<key>([0-9a-z-:]+)<\/key>/gmi.exec(text) || [undefined, undefined];
+	const [, plainClientKey] = /<key>([0-9a-z-:.]+)<\/key>/gmi.exec(text) || [undefined, undefined];
```

Comment:
I _really_ don't recommend we regex to parse XML.  It's one of the major pitfalls of regex and isn't recommended.

Instead, we should use the (Javascript built in XML parser)[https://developer.mozilla.org/en-US/docs/Web/API/DOMParser].

Furthermore, this utility isn't being used except in a test?  But also, I thought all our API/webhooks are JSON and not xml?

Reply:
It is being used [here](https://github.com/atlassian/github-for-jira/blob/fef8a9ebbbec338aaf537b27b91397808bd1357b/src/routes/api/client-key/recover-client-key.ts#L113) to extract the key in api recovery process.

For the DOMParser, I don't think node has it. For simplicity reason, I was origin thinking use a `String.indexOf("<key>")`, but settle on regex.

Reply:
Ah, I'd still use an XML parser and not a regex for this.  xml2json or something so you can just do `data.consumer.key`

Reply:
done.